synapse/storage/data_stores/main/push_rule.py

```diff
@@ -77,6 +80,15 @@ class PushRulesWorkerStore(
     def __init__(self, database: Database, db_conn, hs):
         super(PushRulesWorkerStore, self).__init__(database, db_conn, hs)
 
+        if hs.config.worker.worker_app is None:
+            self._push_rules_stream_id_gen = ChainedIdGenerator(
+                self._stream_id_gen, db_conn, "push_rules_stream", "stream_id"
```

Comment:
I think you probably need to add `EventsWorkerStore` to the superclass list if you want to access _stream_id_gen`

Reply:
Whoops yes